### PR TITLE
47 dropdate nft

### PR DESCRIPTION
**Pull request by JaPilgrim**
-Added Droptime blocker for buying function
-Configured for Kovan Testnet Deployment with truffles & Infura

### CODE REVIEW

File: blockchain/contracts/HelloWorld.sol [hunk 1]
```diff
@@ -25,15 +27,18 @@ contract HelloWorld {
         NFTOwnership memory nftOwnership;
         nftOwnership.nftId = nftHash;
         nftOwnership.owner = nftOwner;
+        nftOwnership.droptime = 1637085033;
```
Comment (BerinikeTech) [inline blockchain/contracts/HelloWorld.sol:30]: This should not be hardcoded but rather be a parameter of the mockNFT function

File: blockchain/contracts/HelloWorld.sol [hunk 2]
```diff
@@ -25,15 +27,18 @@ contract HelloWorld {
...
         nftOwnerships[nftHash] = nftOwnership;
         nfts.push(nftHash);
         return nftHash;
     }
 
-    function buyNFT(uint256 price, uint256 nftHash) public payable {
+    function buyNFT(uint256 price, uint256 nftHash) public payable returns(address) {
+        require(nftOwnerships[nftHash].droptime<=block.timestamp,string(abi.encode("Drop has not started yet! ",(nftOwnerships[nftHash].droptime-block.timestamp),"Seconds left!")));
```
Comment (BerinikeTech) [inline blockchain/contracts/HelloWorld.sol:37]: I think we should change that to not display seconds but hours:minutes:seconds

File: blockchain/contracts/HelloWorld.sol [hunk 3]
```diff
@@ -25,15 +27,18 @@ contract HelloWorld {
...
         price = msg.value;
         nftOwner.transfer(price);
         nftOwnerships[nftHash].owner = buyer;
+        return nftOwnerships[nftHash].owner;
```
Comment (BerinikeTech) [inline blockchain/contracts/HelloWorld.sol:41]: Why did you add this return value?

**Review by BerinikeTech — CHANGES_REQUESTED** [review]

File: blockchain/contracts/HelloWorld.sol (hunk 1 above)
Comment (JaPilgrim) [reply blockchain/contracts/HelloWorld.sol:30]: Good idea!

File: blockchain/contracts/HelloWorld.sol (hunk 3 above)
Comment (JaPilgrim) [reply blockchain/contracts/HelloWorld.sol:41]: Thought it would be beneficial for testing but turns out its not, resolved

File: blockchain/contracts/HelloWorld.sol (hunk 2 above)
Comment (JaPilgrim) [reply blockchain/contracts/HelloWorld.sol:37]: Due to the fact solidity does not have built-in int->String parsing, nor string concatenation(https://ethereum.stackexchange.com/questions/10811/solidity-concatenate-uint-into-a-string), I would rather not implement any display of remaining time yet, as my approach didnt work anyways. Also the result would only be displayed in the blockchain-viewer like etherscan as far as I know & it should rather be implemented in the frontend in the future.

Comment (BerinikeTech) [reply blockchain/contracts/HelloWorld.sol]: Fine with me :)

**Review by BerinikeTech — APPROVED** [review]
Changes were implemented as discussed
